pkg/registry/generic/etcd/etcd.go

```diff
@@ -459,16 +460,46 @@ func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, lis
 	if err != nil {
 		return nil, err
 	}
-	for _, item := range items {
-		accessor, err := meta.Accessor(item)
-		if err != nil {
-			return nil, err
-		}
-		if _, err := e.Delete(ctx, accessor.GetName(), options); err != nil {
-			return nil, err
-		}
+	// Spawn a number of goroutines, so that each one is supposed to delete
+	// at most maxElemsPerWorker elements and they are running in parallel.
+	expectedMaxElemsPerWorker := 1000
```

Comment:
yeah, probably need a max... we already have fan-out on namespace -> resource type... probably want to cap additional fan out on resource type -> workers